03-GettingStarted/06-http-streaming/README.md

```diff
@@ -0,0 +1,427 @@
+
+- **stdio**: Standard input/output, suitable for local and CLI-based tools. Simple but not suitable for web or cloud.
+- **SSE (Server-Sent Events)**: Allows servers to push real-time updates to clients over HTTP. Good for web UIs, but limited in scalability and flexibility.
+- **Streamable HTTP**: Modern HTTP-based streaming transport, supporting notifications and better scalability. Recommended for most production and cloud scenarios.
+
+#### Comparison Table
+| Transport         | Real-time Updates | Streaming | Scalability | Use Case                |
+|-------------------|------------------|-----------|-------------|-------------------------|
+| stdio             | No               | No        | Low         | Local CLI tools         |
+| SSE               | Yes              | Yes       | Medium      | Web, real-time updates  |
+| Streamable HTTP   | Yes              | Yes       | High        | Cloud, multi-client     |
+
+> **Tip:** Choosing the right transport impacts performance, scalability, and user experience. **Streamable HTTP** is recommended for modern, scalable, and cloud-ready applications.
+
+---
+
+## 2. Streaming: Concepts and Motivation
+
+**Streaming** is a technique in network programming that allows data to be sent and received in small, manageable chunks or as a sequence of events, rather than waiting for an entire response to be ready. This is especially useful for:
+
+- Large files or datasets
+- Real-time updates (e.g., chat, progress bars)
+- Long-running computations where you want to keep the user informed
+
+#### Key Concepts
+- Data is delivered progressively, not all at once
+- The client can process data as it arrives
+- Reduces perceived latency and improves user experience
+
+#### Why use streaming?
+- Users get feedback immediately, not just at the end
+- Enables real-time applications and responsive UIs
+- More efficient use of network and compute resources
+
+##### Simple Example: HTTP Streaming Server & Client
+
+**Server (Python, using FastAPI and StreamingResponse):**
+```python
+from fastapi import FastAPI
+from fastapi.responses import StreamingResponse
+import time
+
+app = FastAPI()
+
+async def event_stream():
+    for i in range(1, 6):
+        yield f"data: Message {i}\n\n"
+        time.sleep(1)
+
+@app.get("/stream")
+def stream():
+    return StreamingResponse(event_stream(), media_type="text/event-stream")
+```
+
+**Client (Python, using requests):**
+```python
+import requests
+
+with requests.get("http://localhost:8000/stream", stream=True) as r:
+    for line in r.iter_lines():
+        if line:
+            print(line.decode())
+```
+
+This example demonstrates a server sending a series of messages to the client as they become available, rather than waiting for all messages to be ready.
+
+**How it works:**
+- The server yields each message as it is ready.
+- The client receives and prints each chunk as it arrives.
+
+**Requirements:**
```

Comment:
Move this up to be the lead in sentences for this header, it's good content. Write it as paragraphs rather than bullet list